docs/src/pipe.rst

```diff
@@ -17,16 +17,48 @@ Data types
 
     Pipe handle type.
 
+.. c:type:: uv_credentials_t
+
+    Used to retrieve the process ID and effective user and group IDs
+    from a connected pipe stream.
+
+    ::
+
+        typedef struct {
+            uv_pid_t pid;
+            long uid;
+            long gid;
+        } uv_credentials_t;
+
 
 Public members
 ^^^^^^^^^^^^^^
-
 .. c:member:: int uv_pipe_t.ipc
 
     Whether this pipe is suitable for handle passing between processes.
 
 .. seealso:: The :c:type:`uv_stream_t` members also apply.
 
+.. c:member:: uv_pid_t uv_credentials_t.pid
+
+    The process ID (PID) belonging to the process that initiatied
+    the pipe connection.
+
+.. c:member:: long uv_credentials_t.uid
+.. c:member:: long uv_credentials_t.gid
+
+    The effective user ID (UID) and group ID (GID) under which the
```

Comment:
Since they are the *effective* ones, how about naming the fields `euid` and `egid`? Maybe it can help avoid potential confusion.

Reply:
Hehe, they were before but figured I'd get the opposite critique. I'll switch them back :)